homeassistant/components/esphome/media_player.py

```diff
@@ -70,19 +77,27 @@ def _on_static_info_update(self, static_info: EntityInfo) -> None:
     @esphome_state_property
     def state(self) -> MediaPlayerState | None:
         """Return current state."""
-        return _STATES.from_esphome(self._state.state)
+        state = _STATES.from_esphome(self._state.state)
+        if state is not None:
+            return cast(MediaPlayerState, state)
+
+        return None
 
     @property
     @esphome_state_property
     def is_volume_muted(self) -> bool:
         """Return true if volume is muted."""
-        return self._state.muted
+        return self._state.muted is True
 
     @property
     @esphome_state_property
     def volume_level(self) -> float | None:
         """Volume level of the media player (0..1)."""
-        return self._state.volume
+        volume = self._state.volume
```

Comment:
Where are these changes coming from ?

Reply:
Must be new type checking rules. I couldn't finish the commit without the changes.

Reply:
After rebasing, the complaints went away!